src/proof/nova.rs

```diff
@@ -241,110 +241,104 @@ impl<F: LurkField> Nova<F> for NovaProver<F> {
 #[cfg(test)]
 mod tests {
     use super::*;
-    use crate::eval::{empty_sym_env, Status};
+    use crate::eval::empty_sym_env;
     use crate::proof::{verify_sequential_css, SequentialCS};
-    use crate::writer::Write;
+    use crate::store::ContPtr;
 
     use bellperson::util_cs::{metric_cs::MetricCS, Comparable, Delta};
     use pallas::Scalar as Fr;
 
-    const DEFAULT_CHECK_NOVA: bool = false;
-
     const DEFAULT_CHUNK_FRAME_COUNT: usize = 5;
 
-    fn outer_prove_aux<Fo: Fn(&'_ mut Store<Fr>) -> Ptr<Fr>>(
-        source: &str,
-        expected_result: Fo,
-        expected_cont: Status,
+    fn nova_test_aux(
+        s: &mut Store<Fr>,
+        expr: &str,
+        expected_result: Option<Ptr<Fr>>,
+        expected_env: Option<Ptr<Fr>>,
+        expected_cont: Option<ContPtr<Fr>>,
+        expected_emitted: Option<Vec<Ptr<Fr>>>,
         expected_iterations: usize,
-        chunk_frame_count: usize,
-        check_nova: bool,
-        check_constraint_systems: bool,
-        limit: usize,
-        debug: bool,
     ) {
-        let mut s = Store::default();
-        let expected_result = expected_result(&mut s);
+        nova_test_full_aux(
+            s,
+            expr,
+            expected_result,
+            expected_env,
+            expected_cont,
+            expected_emitted,
+            expected_iterations,
+            DEFAULT_CHUNK_FRAME_COUNT,
+        )
+    }
 
-        let expr = s.read(source).unwrap();
+    fn nova_test_full_aux(
+        s: &mut Store<Fr>,
+        expr: &str,
+        expected_result: Option<Ptr<Fr>>,
+        expected_env: Option<Ptr<Fr>>,
+        expected_cont: Option<ContPtr<Fr>>,
+        expected_emitted: Option<Vec<Ptr<Fr>>>,
+        expected_iterations: usize,
+        chunk_frame_count: usize,
+    ) {
+        let limit = 100000;
+        let expr = s.read(expr).unwrap();
 
         let e = empty_sym_env(&s);
 
         let nova_prover = NovaProver::<Fr>::new(chunk_frame_count);
-        let proof_results = if check_nova {
```

Comment:
This was an important check — allowing tests to opt out of generating a real Nova proof. That was valuable because we are actually using these 'Nova' tests to check constraint system satisfaction, and only running a few real Nova proofs. We should also refactor that for clarity, eventually. Meanwhile, we do need to avoid actually generating Nova proofs for every test case.

Reply:
This change was reverted.